scipy/sparse/_compressed.py

```diff
@@ -1407,6 +1407,48 @@ def _divide_sparse(self, other):
             return out
 
 
+    def broadcast_to(self, shape):
+        if self.format != 'csr':
+            raise NotImplementedError('broadcasting not implemented for'
+                                      'CSC format.')
+        
+        old_shape = self.shape
+        
+        if old_shape == shape:
+            return self.copy()
+        
+        if len(shape) != 2:
+            raise ValueError("Target shape must be a tuple of length 2.")
+
+        # Ensure the old shape can be broadcast to the new shape
+        if any((o != 1 and o != n) for o, n in zip(old_shape, shape)):
+            raise ValueError(f'current shape {old_shape} cannot be'
+                             f' broadcast to new shape {shape}')
+
+        if self.nnz == 0: # array has no non zero elements
+            return  self.__class__(np.zeros(shape, dtype=self.dtype))
+        
+        if old_shape[0] == 1 and old_shape[1] == 1:
+            # Broadcast a single element to the entire shape
+            data = np.full(shape[0] * shape[1], self.data[0])
```

Comment:
Note: even for a 1x1 sparse array, we might have `len(self.data) > 1` due to non-canonical formats (i.e., duplicate entries).

I recommend calling `self.sum_duplicates()` after the `nnz == 0` check.

Reply:
Added at https://github.com/scipy/scipy/pull/21412/commits/188c598e178541096d39ee044ecc20b12581de2d.